
From: Michael Hunold <m.hunold@gmx.de>

i2c_add_driver() may actually fail, but my driver returns 0 regardless. 
Thanks to Arthur Othieno <a.othieno@bluewin.ch> for this obviously correct
patch.

Signed-off-by: Michael Hunold <hunold@linuxtv.org>
Signed-off-by: Arthur Othieno <a.othieno@bluewin.ch>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/media/video/tda9840.c |    8 +++-----
 1 files changed, 3 insertions(+), 5 deletions(-)

diff -puN drivers/media/video/tda9840.c~drivers-media-video-tda9840c-honour-return-code-of drivers/media/video/tda9840.c
--- 25/drivers/media/video/tda9840.c~drivers-media-video-tda9840c-honour-return-code-of	2004-06-22 11:11:28.861312360 -0700
+++ 25-akpm/drivers/media/video/tda9840.c	2004-06-22 11:12:32.592623728 -0700
@@ -268,13 +268,12 @@ static struct i2c_driver driver = {
         .command	= tda9840_command,
 };
 
-static int tda9840_init_module(void)
+static int __init tda9840_init_module(void)
 {
-        i2c_add_driver(&driver);
-        return 0;
+        return i2c_add_driver(&driver);
 }
 
-static void tda9840_cleanup_module(void)
+static void __exit tda9840_cleanup_module(void)
 {
         i2c_del_driver(&driver);
 }
@@ -285,4 +284,3 @@ module_exit(tda9840_cleanup_module);
 MODULE_AUTHOR("Michael Hunold <michael@mihu.de>");
 MODULE_DESCRIPTION("tda9840 driver");
 MODULE_LICENSE("GPL");
-
_
